### PR TITLE
8.0 Add module product_analytic

### PR DESCRIPTION
**Pull request by alexis-via**
Please take a look at the README file to know more about this module and why I developped it.

This PR was initially in OCA/product-attribute https://github.com/OCA/product-attribute/pull/113 and was moved here as suggested by @pedrobaeza  (and the module was renamed).

### CODE REVIEW

File: product_analytic/README.rst [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+   :alt: License: AGPL-3
+
+================
+Product Analytic
+================
+
+This module allows you to configure an **income analytic account** and an **expense analytic account** on products and on product categories. When you select the product in an invoice line, it will check if this product has an income analytic account (for customer invoice/refunds) or an expense analytic account (for supplier invoice/refunds) ; if it doesn't find any, it checks if the category of the product has an income or expense analytic account ; if an analytic account is found, it will be set by default on the invoice line.
+
+This module is an alternative to the official module *account_analytic_default*. The advantages of this module are:
+- it only depends on the *account* module, whereas the *account_analytic_default* module depends on *sale_stock* ;
```
Comment (elicoidal) [inline product_analytic/README.rst]: Use rst list

File: product_analytic/README.rst [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+------------
+
+* Alexis de Lattre <alexis.delattre@akretion.com>
+
+Maintainer
+----------
+
+.. image:: http://odoo-community.org/logo.png
+   :alt: Odoo Community Association
+   :target: http://odoo-community.org
```
Comment (elicoidal) [inline product_analytic/README.rst]: https (x2)

File: product_analytic/account_invoice.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
+# -*- coding: utf-8 -*-
```
Comment (elicoidal) [inline product_analytic/account_invoice.py:1]: Please use simple headers following OCA templates
https://github.com/OCA/maintainer-tools/blob/master/template/module/__openerp__.py#L3

File: product_analytic/account_invoice.py [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from openerp import models, api
```
Comment (elicoidal) [inline product_analytic/account_invoice.py:6]: Please sort in alphabetical order following: https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#imports

File: product_analytic/product_view.xml [hunk 5]
```diff
@@ -0,0 +1,41 @@
+<?xml version="1.0" encoding="utf-8"?>
+
+<!--
```
Comment (elicoidal) [inline product_analytic/product_view.xml:3]: We usually do not have a license reminder in the xml files.

File: product_analytic/__openerp__.py [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+
+{
+    'name': 'Product Analytic',
+    'version': '0.1',
```
Comment (elicoidal) [inline product_analytic/__openerp__.py]: version 8.0.1.0.0

**Comment by elicoidal** [comment]
@alexis-via Thanks for your contribution!

**Comment by pedrobaeza** [comment]
It would be good to include some tests that exercises the module. What do you think of adding code for putting default analytic values when an invoice line is programatically added/changed and the analytic account is empty? This way, we save possible glue modules for each of the possible invoicing sources.

**Comment by alexis-via** [comment]
@pedrobaeza You are right, we should do that because the onchange on product_id is (unfortunately) not played when the invoice is created from a SO/picking/PO.

**Comment by rafaelbn** [comment]
@antespi take a look here.

**Comment by rafaelbn** [comment]
I tested, when you create an invoice with a product which has defined analytic account it doesn't take it to the invoice line:

File: product_analytic/README.rst [hunk 7]
```diff
@@ -0,0 +1,52 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+   :target: http://www.gnu.org/licenses/agpl-3.0-standalone.html
+   :alt: License: AGPL-3
+
+================
+Product Analytic
+================
+
+This module allows you to configure an **income analytic account** and an **expense analytic account** on products and on product categories. When you select the product in an invoice line, it will check if this product has an income analytic account (for customer invoice/refunds) or an expense analytic account (for supplier invoice/refunds) ; if it doesn't find any, it checks if the category of the product has an income or expense analytic account ; if an analytic account is found, it will be set by default on the invoice line.
```
Comment (pedrobaeza) [inline product_analytic/README.rst]: You can put this in the Usage section, and put here

```
This module allows to define an analytic account at product or category level for using it when creating invoices.
```

And please respect 80 cols limit also in this file.

File: product_analytic/account_invoice.py [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+            self, product, uom_id, qty=0, name='', type='out_invoice',
+            partner_id=False, fposition_id=False, price_unit=False,
+            currency_id=False, company_id=None):
+        res = super(AccountInvoiceLine, self).product_id_change(
+            product, uom_id, qty=qty, name=name, type=type,
+            partner_id=partner_id, fposition_id=fposition_id,
+            price_unit=price_unit,
+            currency_id=currency_id, company_id=company_id)
+        if company_id is not None:
+            company_id = company_id
```
Comment (pedrobaeza) [inline product_analytic/account_invoice.py]: This do nothing

**Comment by rafaelbn** [comment]
Hi @alexis-via , are you going to continue this PR? Let me know please, we can PR you and finish this module. Thanks

**Comment by rafaelbn** [comment]
cc @JavierIniesta

**Comment by rafaelbn** [comment]
This PR is fixed here https://github.com/akretion/account-analytic/pull/1, waiting for @alexis-via . thanks!
cc @antespi

**Comment by rafaelbn** [comment]
Tested in local environment, :+1:

**Comment by JavierIniesta** [comment]
Tested in local environment :+1:

**Comment by pedrobaeza** [comment]
I merge, and added in another commit attributions to Antiun in the manifest.

**Pull request merged**